tools/utils.py

```diff
@@ -107,7 +108,20 @@ def delete_dir(dirname):
   """Delete a directory (if it exists)."""
   if not os.path.exists(dirname):
     return
-  shutil.rmtree(dirname)
+
+  def remove_readonly(func, path, _):
+    # We failed to remove a file, perhaps because it or its parent directory are
+    # readonly. Fix that.
+    path = os.path.abspath(path)
+    parent = os.path.dirname(path)
+    try:
+      os.chmod(path, stat.S_IWRITE | stat.S_IREAD | stat.S_IXUSR)
+      os.chmod(parent, stat.S_IWRITE | stat.S_IREAD | stat.S_IXUSR)
+    except:
+      pass
```

Comment:
Why try/except here?  I can't imagine the chmod could ever fail choud it?  And if it did we would probably want to know about it.

Reply:
`chmod` can fail, even in the shell, I learned just now :smile: If the parent directory lacks permissions, chmodding a file in the directory errors.